Source/BestProjectEver/Private/Core/GameState/BPE_GameState.cpp

```diff
@@ -72,11 +74,47 @@ void ABPE_GameState::OnRep_EnemiesAlive()
 	OnEnemyDeath.Broadcast(EnemiesAlive);
 }
 
+//----------------------------------------------------------------------------------------------------------------------
+void ABPE_GameState::HandleMatchResults()
+{
+	MatchResult = bAreAllEnemiesDead ? EMatchResult::Victory : EMatchResult::Defeated;
+}
+
+//----------------------------------------------------------------------------------------------------------------------
+TArray<APlayerState*> ABPE_GameState::GetTopScoringPlayers()
```

Comment:
This part is a bit complicated. You have a getter that isn't only getting the top-scoring players, but also defining them. You're calling this method on server and client, but still having the `TopScoringPlayers` replicated.

Reply:
![image](https://user-images.githubusercontent.com/85327912/175394818-67bbae93-6635-4f43-956d-510cd7ac0d23.png)
Now only the server is calling this DetermineMatchResults

Reply:
![image](https://user-images.githubusercontent.com/85327912/175395076-d140ccf8-2d76-4dd5-b329-9ba4a9e40c93.png)

Reply:
And TopScoringPlayers array is replicated